Branch-SDK/Branch-SDK/BNCContentDiscoveryManager.m

```diff
@@ -71,61 +71,255 @@ - (NSString *)standardSpotlightIdentifierFromActivity:(NSUserActivity *)userActi
+    [self indexContentWithTitle:title
+                    description:description
+              publiclyIndexable:publiclyIndexable
+                           type:kUTTypeGeneric
+                   thumbnailUrl:thumbnailUrl
+                       keywords:nil
+                       userInfo:userInfo
+                 expirationDate:nil
+                       callback:NULL];
 }
 
-- (void)indexContentWithTitle:(NSString *)title description:(NSString *)description publiclyIndexable:(BOOL)publiclyIndexable thumbnailUrl:(NSURL *)thumbnailUrl keywords:(NSSet *)keywords userInfo:(NSDictionary *)userInfo {
-    [self indexContentWithTitle:title description:description publiclyIndexable:publiclyIndexable type:kUTTypeGeneric thumbnailUrl:thumbnailUrl keywords:keywords userInfo:userInfo expirationDate:nil callback:NULL];
+- (void)indexContentWithTitle:(NSString *)title
+                  description:(NSString *)description
+            publiclyIndexable:(BOOL)publiclyIndexable
+                 thumbnailUrl:(NSURL *)thumbnailUrl
+                     keywords:(NSSet *)keywords
+                     userInfo:(NSDictionary *)userInfo {
+    [self indexContentWithTitle:title
+                    description:description
+              publiclyIndexable:publiclyIndexable
+                           type:kUTTypeGeneric
+                   thumbnailUrl:thumbnailUrl
+                       keywords:keywords
+                       userInfo:userInfo
+                 expirationDate:nil
+                       callback:NULL];
 }
 
-- (void)indexContentWithTitle:(NSString *)title description:(NSString *)description publiclyIndexable:(BOOL)publiclyIndexable type:(NSString *)type thumbnailUrl:(NSURL *)thumbnailUrl keywords:(NSSet *)keywords userInfo:(NSDictionary *)userInfo callback:(callbackWithUrl)callback {
-    [self indexContentWithTitle:title description:description publiclyIndexable:publiclyIndexable type:type thumbnailUrl:thumbnailUrl keywords:keywords userInfo:userInfo expirationDate:nil callback:callback];
+- (void)indexContentWithTitle:(NSString *)title
+                  description:(NSString *)description
+            publiclyIndexable:(BOOL)publiclyIndexable
+                         type:(NSString *)type
+                 thumbnailUrl:(NSURL *)thumbnailUrl
+                     keywords:(NSSet *)keywords
+                     userInfo:(NSDictionary *)userInfo
+               expirationDate:(NSDate*)expirationDate
+                     callback:(callbackWithUrl)callback {
+    [self indexContentWithTitle:title
+                    description:description
+              publiclyIndexable:publiclyIndexable
+                           type:type
+                   thumbnailUrl:thumbnailUrl
+                       keywords:keywords
+                       userInfo:userInfo
+                 expirationDate:nil
+                       callback:callback
+              spotlightCallback:(callbackWithUrlAndSpotlightIdentifier)callback];
```

Comment:
why are we passing in callback 2x? this seems like it would crash. should be `nil`, no?